web/src/ProductSelection.jsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useInstallerClient } from "./context/installer";
+
+import {
+  Button,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  Radio
+} from "@patternfly/react-core";
+
+import {
+  EOS_PRODUCT_SUBSCRIPTIONS as SectionIcon,
+} from "eos-icons-react";
+
+import Layout from "./Layout";
+import Center from "./Center";
+
+export default function ProductSelection() {
+  const client = useInstallerClient();
+  const [previous, setPrevious] = useState(undefined);
+  const [selected, setSelected] = useState(undefined);
+  const [products, setProducts] = useState(undefined);
+
+  useEffect(() => {
+    const loadProducts = async () => {
+      // TODO: check if it can be done in one line without performance penalty
+      const availableProducts = await client.software.getProducts();
+      const selectedProduct = await client.software.getSelectedProduct();
+
+      setProducts(availableProducts);
+      setPrevious(selectedProduct);
+      setSelected(selectedProduct);
+    };
+
+    loadProducts().catch(console.error);
+  }, [client.software]);
+
+  useEffect(() => {
+    // TODO: display a notification in the UI to emphasizes that 
+    // selected product has changed
+    return client.software.onProductChange(setSelected)
+  }, [client.software]);
+
+  if (!products) return <Center>"Loading available products..."</Center>;
+
+  const isSelected = p => p.id === selected;
+
+  const accept = async () => {
+    if (selected !== previous) {
+      // FIXME remove console log
+      console.log(selected, "selected");
+
+      // TODO: handle errors
+      await client.software.selectProduct(selected);
+    }
+  };
+
+  const ContinueButton = () => {
+    return (
+      <Button isLarge variant="primary" onClick={accept}>
+        Continue
+      </Button>
+    );
+  };
+
+  const buildOptions = () => {
+    const options = products.map((p) => (
```

Comment:
One feature that I think is a must have is to skip this selection if there is just one product. I expect it will be common case when customer use his own handcrafted iso for install or wants to have it as PXE entries. Also maybe for some products  we want specialized installers.....Of course second question is how to react with system roles, but it is topic for future.

Reply:
Of course, this component/screen will not be shown when there is only one product.